openfisca_core/taxbenefitsystems.py

```diff
@@ -102,42 +98,115 @@ def new_scenario(self):
     def prefill_cache(self):
         pass
 
+    def load_variable(self, variable_class, update = False):
+        name = unicode(variable_class.__name__)
+        variable_type = variable_class.__bases__[0]
+        attributes = variable_class.__dict__
+
+        existing_column = self.get_column(name)
+
+        if existing_column and not update:
+            # Variables that are dependencies of others (trough a conversion column)can be loaded automatically
+            if name in self.automatically_loaded_variable:
+                self.automatically_loaded_variable.remove(name)
+                return self.get_column(name)
+            raise Exception("Variable {} is already defined. Use `update_variable` to replace it.".format(name))
+
+        if not existing_column and update:
+            raise Exception("Variable {} doesn't exist and can't be updated".format(name))
+
+        # We pass the variable_class just for introspection for parsers.
+        variable = variable_type(name, attributes, variable_class)
+        # We need the tax benefit system to identify columns mentioned by reference or PersonToEntityColumn...
+        column = variable.to_column(self)
+        self.column_by_name[column.name] = column
+
+        return column
+
+    def add_variable(self, variable_class):
+        return self.load_variable(variable_class, update = False)
+
+    def update_variable(self, variable_class):
+        return self.load_variable(variable_class, update = True)
+
+    def add_variables_from_file(self, file):
+        module_name = path.splitext(path.basename(file))[0]
+        module_directory = path.dirname(file)
+        module = load_module(module_name, *find_module(module_name, [module_directory]))
+
+        potential_variables = [getattr(module, c) for c in dir(module) if not c.startswith('__')]
+        for pot_variable in potential_variables:
+            # We only want to get the module classes defined in this module (not imported)
+            if ((isclass(pot_variable) and
+                 issubclass(pot_variable, AbstractVariable) and
+                 pot_variable.__module__.endswith(module_name))):
+                self.add_variable(pot_variable)
+
+    def add_variables_from_directory(self, directory):
+        py_files = glob.glob(path.join(directory, "*.py"))
+        for py_file in py_files:
+            self.add_variables_from_file(py_file)
+        subdirectories = glob.glob(path.join(directory, "*/"))
+        for subdirectory in subdirectories:
+            self.add_variables_from_directory(subdirectory)
+
+    def add_variables(self, *variables):
+        for variable in variables:
+            self.add_variable(variable)
+
+    def load_extension(self, path):
+        if path.endswith('.git'):
+            import tempfile
+            import subprocess
+            temp_dir = tempfile.mkdtemp()
+            clone_command = ' '.join(['git clone', path, temp_dir, '&> /dev/null'])
```

Comment:
Yay I admit this may be a little weird. I thought I would need it, but actually I'm not using it, so I'll just remove the feature. Adding extensions from a package seems interesting too, I'll keep that in mind.